### PR TITLE
VMware NSX-T SDN Install and Config updates

### PR DESCRIPTION
**Pull request by RobbieJVMW**
Updated documentation for VMware NSX-T SDN plug-in. 
Example topology and ansible config. 
Links to full configuration guide following the same style and formatting of existing SDN Plug-In docs. 

Targetting - Enterprise 3.1 / Origin 

@openshift/team-documentation

### CODE REVIEW

**Comment by RobbieJVMW** [comment]
@dav1x

**Comment by vikram-redhat** [comment]
Thanks for submitting this PR. We are not currently accepting partner documentation in our docs. I am going to close this PR. Please work with your partner team at Red Hat for approvals for these updates.

**Comment by RobbieJVMW** [comment]
@vikram-redhat This work has been completed via the partner team and has been approved. 
@dav1x can provide more details.

File: install_config/configuring_nsxtsdn.adoc [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+node03.example.com ansible_ssh_host=192.168.220.7 OpenShift_node_group_name='node-config-compute' OpenShift_ip=192.168.220.7
+node04.example.com ansible_ssh_host=192.168.220.8 OpenShift_node_group_name='node-config-compute' OpenShift_ip=192.168.220.8
+----
+
+Check prerequisites for {product-title} with NSX-T CNI:
+----
+$ ansible-playbook -i hosts openshift-ansible/playbooks/prerequisites.yml
+----
+
+Once the above playbook has executed successfully, run the following on all nodes:
```
Comment (mburke5678) [inline install_config/configuring_nsxtsdn.adoc]: @RobbieJVMW Should this be a prerequisite?

File: install_config/configuring_nsxtsdn.adoc [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+See xref:../architecture/networking/network_plugins.adoc#nsx-sdn[Networking]
+for more information on how NSX-T is integrated with {product-title}.
+
+[[nsx-t-sdn-operations-workflow]]
+== Example Topology
+
+There is a *T0 router* on the top that connects physical with virtual world. We also have *T1 router* acting as a default gateway for the OpenShift VMs.
+
+Those VMs have *two vNICs* each. One vNIC is connected to Management Logical Switch for accessing the VMs. The second vNIC is connected to a dump Logical Switch and is used by nsx-node-agent to uplink the POD networking. For further details, we strongly suggest reading the link:https://docs.vmware.com/en/VMware-NSX-T-Data-Center/2.3/nsxt_23_ncp_OpenShift.pdf[NSX Container Plug-in for OpenShift].
```
Comment (mburke5678) [inline install_config/configuring_nsxtsdn.adoc]: @RobbieJVMW  I get a 404 at that link. Can you confirm the link?

File: install_config/configuring_nsxtsdn.adoc [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+The LoadBalancer used for configuring OpenShift Routes plus all project’s *T1 routers* and *Logical Switches* are created automatically later when we install OpenShift. In this topology we use the default OpenShift HAProxy Router for all infrastructure components such as Grafana, Prometheus, Console, Service Catalog and others.
+
+This means that the DNS records for the infra components need to point the infrastructure node IP addresses since the HAProxy uses the host network namespace.
+This works well for infra routes but in order to avoid exposing the infra nodes management IPs to the outside world we will be deploying application specific routes to the NSX-T LoadBalancer.
+
+The topology here assumes 3 x OpenShift master virtual machines and 4 x OpenShift worker virtual machines(two for infrastructure and two for compute).
```
Comment (mburke5678) [inline install_config/configuring_nsxtsdn.adoc]: @RobbieJVMW Can the user use 4.x and 3.x together? Or are you connecting the two?

File: install_config/configuring_nsxtsdn.adoc [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+image:images/nsxt-transportnodes.png[NSX Transport Nodes]
+
+* DNS requirements:
+** *You must add* to your DNS server a new entry with a wildcard to the 'infra' nodes, this allows load balancing by NSX-T or other third-party LB. In the 'hosts' file below, it's defined by 'openshift_master_default_subdomain' variable.
+** You have also to update your DNS server accordingly with: 'openshift_master_cluster_hostname' & 'openshift_master_cluster_public_hostname' variables.
+
+* Virtual Machine requirements:
+** {product-title} node VMs must have *two vNICs*:
+** Management vNIC connected to the *Logical Switch* that is uplinked to the management *T1 router*.
+** The second vNIC on all VMs needs to have two Tags in *NSX-T* in order 'nsx-container-plugin' (NCP) to know which port needs to be used as a parent VIF for all PODs running in the particular OpenShift node. Tags need to be as following:
```
Comment (mburke5678) [inline install_config/configuring_nsxtsdn.adoc]: @RobbieJVMW In _Tags in *NSX-T* in order_ what does _Tags in NSX-T_ mean?

File: install_config/configuring_nsxtsdn.adoc [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+----
+
+Check prerequisites for {product-title} with NSX-T CNI:
+----
+$ ansible-playbook -i hosts openshift-ansible/playbooks/prerequisites.yml
+----
+
+Once the above playbook has executed successfully, run the following on all nodes:
+
+Assuming NCP Container image is downloaded locally on all nodes.
```
Comment (mburke5678) [inline install_config/configuring_nsxtsdn.adoc]: @RobbieJVMW Does the OpenShift installer download these images or is this a manual step? 

| docker load -i nsx-ncp-rhel-xxx.tar
Is `nsx-ncp-rhel-xxx` a literal string to enter or does the user need to know the file name? If the later, how does the user know the file name?

File: install_config/configuring_nsxtsdn.adoc [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+nsx-node-agent-cldks   2/2       Running   0          2h        192.168.220.8   node04.example.com     <none>
+nsx-node-agent-m2p5l   2/2       Running   28         3h        192.168.220.4   master03.example.com   <none>
+nsx-node-agent-pcfd5   2/2       Running   0          1h        192.168.220.7   node03.example.com     <none>
+nsx-node-agent-ptwnq   2/2       Running   26         3h        192.168.220.2   master01.example.com   <none>
+nsx-node-agent-xgh5q   2/2       Running   26         3h        192.168.220.3   master02.example.com   <none>
+----
+
+== Check NSX-T after {product-title} deployment
+
+Check routing. T1 routers has been created during namespace creation and linked to T0 router:
```
Comment (mburke5678) [inline install_config/configuring_nsxtsdn.adoc]: @RobbieJVMW 

> T1 router has been created during namespace creation

Is it important when the routers were created? Is saying _T1 router has been created_

File: install_config/configuring_nsxtsdn.adoc [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+image:images/nsxt-routing.png[NSX routing]
+
+Observe Network Traceflow and visibility, for example between 'console' and 'grafana'. +
+To help fully understand, secure and optimise communications between pods, projects, virtual machines and external services: +
+image:images/nsxt-visibility.png[NSX visibility]
+
+Check Load Balancing. NSX-T offers Load Balancer and Ingress Controller capabilities:
+image:images/nsxt-loadbalancing.png[NSX loadbalancing]
```
Comment (mburke5678) [inline install_config/configuring_nsxtsdn.adoc]: @RobbieJVMW Is the `nsxt-loadbalancing.png` image correct? The **Server Pools** tab of the **Load Balancing** page is displayed. Should this be the **Load Balancers** tab?

File: install_config/configuring_nsxtsdn.adoc [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+
+For additional configuration and options please refer to the link:https://docs.vmware.com/en/VMware-NSX-T-Data-Center/2.4/rn/NSX-Container-Plugin-Release-Notes.html[VMware NSX-T v2.4 OpenShift Plug-In] documentation.
```
Comment (mburke5678) [inline install_config/configuring_nsxtsdn.adoc]: @RobbieJVMW This link goes to the release notes. Is that the best place to find _configurations and options_?

File: install_config/configuring_nsxtsdn.adoc (hunk 2 above)
Comment (RobbieJVMW) [reply install_config/configuring_nsxtsdn.adoc]: The link should be : https://docs.vmware.com/en/VMware-NSX-T-Data-Center/2.4/nsxt_24_ncp_openshift.pdf

If you could check it works for you, its fine for me.

File: install_config/configuring_nsxtsdn.adoc (hunk 3 above)
Comment (RobbieJVMW) [reply install_config/configuring_nsxtsdn.adoc]: This is referring to the number of virtual machines not the Openshift version ( if that's the question).  The target for this update is 3.11

File: install_config/configuring_nsxtsdn.adoc (hunk 4 above)
Comment (RobbieJVMW) [reply install_config/configuring_nsxtsdn.adoc]: Tag is essentially a label. NSX Tag's are how the nsx platform manages relationships between components.

File: install_config/configuring_nsxtsdn.adoc (hunk 5 above)
Comment (RobbieJVMW) [reply install_config/configuring_nsxtsdn.adoc]: This is a manual set with 'nsx-ncp-rhel-xxx' where xxx refers to the NCP build version.

File: install_config/configuring_nsxtsdn.adoc (hunk 4 above)
Comment (mburke5678) [reply install_config/configuring_nsxtsdn.adoc]: I saw that better when I got the image to appear in my preview. Thanks!

File: install_config/configuring_nsxtsdn.adoc (hunk 1 above)
Comment (dav1x) [reply install_config/configuring_nsxtsdn.adoc]: Howdy @mburke5678 ! Yes, this prepares the nodes for OCP installation. See:

https://github.com/openshift/openshift-ansible/tree/release-3.11#simple-all-in-one-localhost-installation

**Pull request closed without merging**